agentos/registry.py

```diff
@@ -83,6 +84,54 @@ def from_repo(
         ), "YAML is the only registry file format supported currently"
         return cls.from_yaml(repo.get_local_repo_dir(version) / file_path)
 
+    @classmethod
+    def from_repo_inferred(
+        cls,
+        repo: "Repo",
+        py_file_suffixes: Sequence[str] = [".py", ".python"],
+        requirements_file: str = "requirements.txt",
+    ):
+        from agentos.component import Component  # Avoid circular ref.
+
+        reg = InMemoryRegistry()
+        # get list of python files in Repo
+        py_files = set()
+        for suff in py_file_suffixes:
+            found = repo.get_local_repo_dir().rglob(f"*{suff}")
```

Comment:
~~Does rglob recursively search in directories?~~  Answered my own question by running through the demo.